LECTURE 7

In [1]:
report_card_dictionary = {"student_names" : ["zaid", "jay", "aryan", "sahil", "riya"],
                          "math_scores" : [50, 43,48,45,31],
                          "chemistry_scores" : [23, 43,68,85,11],
                          "information_technology_scores" : [51,33,58,25,81],
                          "physics_scores" : [60, 73,28,90,21]}
report_card_dictionary

{'student_names': ['zaid', 'jay', 'aryan', 'sahil', 'riya'],
 'math_scores': [50, 43, 48, 45, 31],
 'chemistry_scores': [23, 43, 68, 85, 11],
 'information_technology_scores': [51, 33, 58, 25, 81],
 'physics_scores': [60, 73, 28, 90, 21]}

In [2]:
#setup
import pandas as pd

In [3]:
#converting dictionary to dataframe
report_card_df = pd.DataFrame(report_card_dictionary)
report_card_df

student_names  math_scores  chemistry_scores  information_technology_scores  \
0          zaid           50                23                             51   
1           jay           43                43                             33   
2         aryan           48                68                             58   
3         sahil           45                85                             25   
4          riya           31                11                             81   

   physics_scores  
0              60  
1              73  
2              28  
3              90  
4              21

In [4]:
#showing only 10 rows using head function
report_card_df.head()

student_names  math_scores  chemistry_scores  information_technology_scores  \
0          zaid           50                23                             51   
1           jay           43                43                             33   
2         aryan           48                68                             58   
3         sahil           45                85                             25   
4          riya           31                11                             81   

   physics_scores  
0              60  
1              73  
2              28  
3              90  
4              21

In [5]:
report_card_df.head(2)

student_names  math_scores  chemistry_scores  information_technology_scores  \
0          zaid           50                23                             51   
1           jay           43                43                             33   

   physics_scores  
0              60  
1              73

In [6]:
#sample of 2 rows in our data frame
report_card_df.sample(2)

student_names  math_scores  chemistry_scores  information_technology_scores  \
4          riya           31                11                             81   
0          zaid           50                23                             51   

   physics_scores  
4              21  
0              60

In [7]:
report_card_df.describe()

math_scores  chemistry_scores  information_technology_scores  \
count     5.000000          5.000000                       5.000000   
mean     43.400000         46.000000                      49.600000   
std       7.436397         30.692019                      22.018174   
min      31.000000         11.000000                      25.000000   
25%      43.000000         23.000000                      33.000000   
50%      45.000000         43.000000                      51.000000   
75%      48.000000         68.000000                      58.000000   
max      50.000000         85.000000                      81.000000   

       physics_scores  
count         5.00000  
mean         54.40000  
std          29.39898  
min          21.00000  
25%          28.00000  
50%          60.00000  
75%          73.00000  
max          90.00000

In [8]:
#create a new column called total score
report_card_df["total_score"] = report_card_df["math_scores"] + report_card_df["chemistry_scores"] + report_card_df["information_technology_scores"] + report_card_df["physics_scores"]
report_card_df.head()

student_names  math_scores  chemistry_scores  information_technology_scores  \
0          zaid           50                23                             51   
1           jay           43                43                             33   
2         aryan           48                68                             58   
3         sahil           45                85                             25   
4          riya           31                11                             81   

   physics_scores  total_score  
0              60          184  
1              73          192  
2              28          202  
3              90          245  
4              21          144

In [9]:
#create a colmun called average score
report_card_df["average_score"] = report_card_df["total_score"]/4
report_card_df.head()

student_names  math_scores  chemistry_scores  information_technology_scores  \
0          zaid           50                23                             51   
1           jay           43                43                             33   
2         aryan           48                68                             58   
3         sahil           45                85                             25   
4          riya           31                11                             81   

   physics_scores  total_score  average_score  
0              60          184          46.00  
1              73          192          48.00  
2              28          202          50.50  
3              90          245          61.25  
4              21          144          36.00

In [10]:
report_card_df.describe()

math_scores  chemistry_scores  information_technology_scores  \
count     5.000000          5.000000                       5.000000   
mean     43.400000         46.000000                      49.600000   
std       7.436397         30.692019                      22.018174   
min      31.000000         11.000000                      25.000000   
25%      43.000000         23.000000                      33.000000   
50%      45.000000         43.000000                      51.000000   
75%      48.000000         68.000000                      58.000000   
max      50.000000         85.000000                      81.000000   

       physics_scores  total_score  average_score  
count         5.00000     5.000000       5.000000  
mean         54.40000   193.400000      48.350000  
std          29.39898    36.287739       9.071935  
min          21.00000   144.000000      36.000000  
25%          28.00000   184.000000      46.000000  
50%          60.00000   192.000000      48.000000  
75%          73.00000   202.000000      50.500000  
max          90.00000   245.000000      61.250000

INSIGHTS
what is the mean chemistry score = 46
what is the lowest physics score = 21
what is the highest total score = 245
what is the mean total score = 193.4
what is the mean average score = 48.35

In [11]:
# CREATING A TABLE CALLED REPORT CARD INSIGHTS
report_card_insights_df = report_card_df.describe().transpose()
report_card_insights_df

count    mean        std    min    25%    50%  \
math_scores                      5.0   43.40   7.436397   31.0   43.0   45.0   
chemistry_scores                 5.0   46.00  30.692019   11.0   23.0   43.0   
information_technology_scores    5.0   49.60  22.018174   25.0   33.0   51.0   
physics_scores                   5.0   54.40  29.398980   21.0   28.0   60.0   
total_score                      5.0  193.40  36.287739  144.0  184.0  192.0   
average_score                    5.0   48.35   9.071935   36.0   46.0   48.0   

                                 75%     max  
math_scores                     48.0   50.00  
chemistry_scores                68.0   85.00  
information_technology_scores   58.0   81.00  
physics_scores                  73.0   90.00  
total_score                    202.0  245.00  
average_score                   50.5   61.25

In [12]:
#creating a csv file of our report card insights

report_card_insights_df.to_csv("report card insights.csv")